cep-9999.md

```diff
@@ -0,0 +1,94 @@
+# CEP XXXX - Names in conda packages and channels
+
+<table>
+<tr><td> Title </td><td> CEP XXXX - Names in conda packages and channels </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt; <br /> Matthew R. Becker &lt;becker.mr@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Mar 11, 2025</td></tr>
+<tr><td> Updated </td><td> Mar 11, 2025</td></tr>
+<tr><td> Discussion </td><td> N/A </td></tr>
+<tr><td> Implementation </td><td> N/A </td></tr>
+</table>
+
+## Abstract
+
+This CEP aims to standardize names and other strings used throughout the conda ecosystem.
+
+## Specification
+
+### Artifacts
+
+The conda ecosystem distinguishes between two types of packages:
+
+- Distributable package names: represented by a concrete, downloadable, extractable conda artifact.
+- Virtual package names: not backed by any physical artifact. They only exist on the client side.
+
+### Package names
+
+A distributable package name MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. It MUST start with a letter, a number, or a single underscore. It MUST NOT include two consecutive separators (hyphen, period, underscore).
+
+Virtual package names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. They MUST NOT use two consecutive separators, with one exception: they MUST start with two underscores.
+
+This means that distributable package names MUST match the following case-insensitive regex: `^(([a-z0-9])|([a-z0-9_](?!_)))[._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+Virtual package names MUST follow this other regex: `^__[a-z0-9][._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+In all cases, the maximum length of a package name MUST NOT exceed 128 characters.
+
+### Version strings
+
+Version strings MUST only consist of digits, periods, lowercase ASCII letters, underscores, plus
+signs and exclamation marks. Additional rules apply but are out of scope in this CEP and will be
+discussed separately.
+
+The maximum length of a version string MUST NOT exceed 128 characters.
+
+### Build strings
+
+Builds strings MUST only consist of ASCII letters, numbers, periods, and underscores. They MUST match this regex `^[a-zA-Z0-9_\.]+$`.
+
+The maximum length of a build string MUST NOT exceed 128 characters.
+
+### Filenames
+
+Regular conda artifacts MUST be named following this scheme:
+
+```
+<package name>-<version string>-<build string>.<extension>
+```
+
+Virtual conda packages SHOULD NOT need filename standardization.
+
+### Channels
+
+#### Channel names
+
+A channel name is defined as the
+
+Channel names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. They MUST start and end with a letter or a digit. They MUST NOT contain two consecutive separators (hyphen, underscore, period). In other words, they MUST match this regex:  `^[a-z0-9]+((-|_|.)[a-z0-9]+)*$`.
```

Comment:
As of 2025-03-12T19:00Z, out of 1,918,426 channel/org/names on anaconda.org:

* 7,219 violate the regex `^[a-z0-9]+((-|_|.)[a-z0-9]+)*$`;
* 98 violate the regex `^[a-z0-9][a-z0-9_.-]*$` (allowing channel names to end with `_`, `.`, or `-`); and
* 6 violate `^[a-z0-9_][a-z0-9_.-]*$` (allowing channel names to start with `_`)

Of those six, five start with `.`, and the other starts with `~`. Which I would claim, we want to disallow in any case.

Reply:
Can you provide statistics on usage of the channels in each of those cases? 

I'd very much like to leave behind channels that start with an `_`, and I want to know how destructive that decision will be.

Reply:
> Can you provide statistics on usage of the channels in each of those cases?

Will likely take me some time to dig into that. Since channel names and user names are roughly equivalent .org, it's not the easiest task to separate "logged traffic due to using this as a conda channel" from "logged traffic because someone logged in and did soemthing else".